pymedphys/_experimental/pseudonymisation/__init__.py

```diff
@@ -134,3 +138,67 @@ def is_valid_strategy_for_keywords(
         if vr not in replacement_strategy:
             return False
     return True
+
+
+def pseudonymise(dicom_input, output_path=None):
+    """Convenient API to pseudonymisation.
+    Elements whose tags are not in the pydicom dictionary will be deleted
+    PatientSex will not be modified/pseudonymised
+    For fine tune control, use anonymise_dataset() instead
+
+    Parameters
+    ----------
+    dicom_input : ``pydicom.dataset.Dataset | str | pathlib.Path``
+        Either a dataset, a path to a file or a path to a directory
+    output_path : ``str | pathlib.Path``, optional
+        If the input is a file or a path, the directory to place the
+        pseudonymised files, by default None
+
+    Returns
+    -------
+    ``pydicom.dataset.Dataset`` | ``str`` | ``list`` of ``str``
+        if the dicom_input was a dataset, return the pseudonymised dataset
+        if the dicom input was a file, return the path to the pseudonymised file.
+        if the dicom input was a directory, return the list of successfully
+        anonymised files, and return that instead of None
```

Comment:
I'm undecided here about having the one function run off and do different things. I understand the approach though. Part of me wonders if it would be better to expose dataset input/output only, and then provide example code within the docstring that can glob through an input directory and save the datsets to an output directory... As in don't expose the directory code via the library API, just provide someone with the machinery to do the file globbing themselves.

At the end of the day, if this does the "psudononymise a pydicom dataset" well, then the rest of the "which datasets to run this on" logic can be implemented with user code.

I'm not sure about this though. I'm more running on intuition here and would be more than happy for you to say, "this is the way I think it works best".

Reply:
There is enough involved in operating on a file or on a directory that is in common for any user or programmer that it warrants providing the functionality to the user/programmer.
Note that the exposed functionality through the CLI is "file or directory", and the CLI for it leaves it to the implementation to sort out if it's a file or a directory.
I've mentioned elsewhere that file and directory operation should be exposed.

Further, this is acting as polymorphism, whose value, IMHO is to reduce cognitive overload.
So it achieves exposing the file and directory operation without exposing a variety of methods that the user probably doesn't need to know.

As a user/programmer, it's nice if I can just say "here's my data, go deal with it".
Which is the long winded way of saying 
"this is the way I think it works best"

Reply:
kk :)